### PR TITLE
SDA name change

### PR DESCRIPTION
**Pull request by helge22a**
## Description

Reverting commits made in PR #2927 to remove "." from assim.sequential package. @infotroph @mdietze @serbinsh @DongchenZ @Qianyuxuan

I am not sure why GitHub is including other commits not related to ones I just made on this new branch, but I cannot figure out how to get rid of them?

## Motivation and Context

Nimble throughs an error with package names containing "."
## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of CITATION.cff
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/assim.sequential/inst/WillowCreek/SDA_Workflow_LAI.R [hunk 1]
```diff
@@ -0,0 +1,345 @@
+# ----------------------------------------------------------------------
```
Comment (mdietze) [inline modules/assim.sequential/inst/WillowCreek/SDA_Workflow_LAI.R:1]: This seems to be an entirely new file that you're adding as part of the PR. Would prefer if you hold this back for a separate PR

File: modules/data.land/R/cohort2pool.R [hunk 2]
```diff
@@ -8,14 +8,6 @@
...
 ##' @param dat veg_info file
 ##' @param allom_param parameters for allometric equation, a and b. Based on base-10 log-log linear model (power law)
-##'
-##' @author Saloni Shah
-##' @examples
-##' \dontrun{
-##' veg_file <- "~/downloads/FFT_site_1-25665/FFT.2008.veg.rds"
-##' cohort2pool(veg_File = veg_file, allom_param = NULL)
-##' }
-##'
```
Comment (mdietze) [inline modules/data.land/R/cohort2pool.R:18]: 1. seems unrelated to current PR
2. Not sure why you are removing the example and deleting Saloni's authorship

File: modules/data.land/R/extract_veg.R [hunk 3]
```diff
@@ -67,14 +67,17 @@ extract_veg <- function(new_site, start_date, end_date,
...
+ } 
+ if(source == "NEON_veg"){
+   format_name = "usda"
+   code_col = "species_USDA_symbol"
    }else{
       code_col    <- "species_USDA_symbol"
       format_name <- "usda"
       obs[obs$species_USDA_symbol != "2PLANT", ] #removes the rows with 2PLANT, this is a NEON specific code that means they could not identify the species 
    }
+>>>>>>> f5194f84b26a85bbfcb279cc4665d7560205efc7
```
Comment (mdietze) [inline modules/data.land/R/extract_veg.R:79]: This is indicative of a merge conflict and needs to be resolved

File: modules/assim.sequential/inst/WillowCreek/SDA_Workflow_LAI.R (hunk 1 above)
Comment (helge22a) [reply modules/assim.sequential/inst/WillowCreek/SDA_Workflow_LAI.R:1]: Yes, sorry that was a mistake it must've got added when I was switching branches because I still had the script open. I will remove it.

File: modules/data.land/R/cohort2pool.R (hunk 2 above)
Comment (helge22a) [reply modules/data.land/R/cohort2pool.R:18]: Yes, I am going to revert this commit

File: modules/data.land/R/extract_veg.R (hunk 3 above)
Comment (helge22a) [reply modules/data.land/R/extract_veg.R:79]: Yes I am going to revert this commit

**Review by mdietze — APPROVED** [review]

**Comment by mdietze** [comment]
noting from PEcAn meeting that @infotroph needs to check the impacts of this change on make, etc before pulling.

**Review by infotroph — CHANGES_REQUESTED** [review]
* Please add a changelog entry announcing the package name change
* The package name also needs to be changed in at least the following places (Found via `grep -RFn 'assim.sequential' pecan/`; all hits that are referring to the path `modules/assim.sequential/` can stay, all that mean the package name should be changed):
    - shiny/SDAdashboard/server.R (lines 175 and 177)
    - All references to the package name in function documentation inside the package, e.g. `sda_plotting.R` line 4 and `sda.enkf_refactored.R` line 165 (TBH both these lines could use rephrasing and would probably be clearer if they didn't contain the package name at all, but that should be a different PR -- for today we just want the documentation to stay consistent)
    - Ditto in function documentation of other packages. The only one I saw was line 3 of `modules/uncertainty/R/prep.data.assim.R`, but please confirm that.
    - There are many remaining function calls throughout the subfolders of `inst/` that still use `PEcAn.assim.sequential::`. For archived files like `sda_backup` I'm always of two minds about whether to update them, but since you've already updated the `library(...)` usages in these please update the others as well.

**Comment by infotroph** [comment]
Places I checked for build tool assumptions that might break on this name change:
* [x] Package version reporting functions in `PEcAn.all`? ✅: They match any package with "PEcAn" in its name and don't check for the dot
* [x] `Makefile`? ✅: Works entirely on directory names, not package names
* [x] `scripts/` folder? I manually reviewed all contents to confirm...
    - [x] Most ✅: don't care about package names
    - [x] EFI scripts: ✅: are already updated to use new name
    - [ ] generate_dependencies.R: ❌ works only on packages whose name starts with a literal `PEcAn.` I believe this can be fixed by changing `"^PEcAn."` to `"^PEcAn"` on lines 53 and 60.
 * [x] `.github/workflows`? ✅: all the package-specific steps are handled by Make, which I checked already. 
 * [x] `docker/`? ✅: all the package-specific steps are handled by Make, which I checked already.

File: modules/uncertainty/DESCRIPTION [hunk 4]
```diff
@@ -43,7 +43,9 @@ Imports:
     plyr (>= 1.8.4),
     purrr,
     randtoolbox,
-    rlang
+    rlang,
+    PEcAn.data.atmosphere
```
Comment (mdietze) [inline modules/uncertainty/DESCRIPTION:47]:
```suggestion
    PEcAn.data.atmosphere,
```

File: modules/uncertainty/DESCRIPTION [hunk 5]
```diff
@@ -43,7 +43,9 @@ Imports:
     plyr (>= 1.8.4),
     purrr,
     randtoolbox,
-    rlang
+    rlang,
+    PEcAn.data.atmosphere,
+    PEcAnAssimSequential
```
Comment (infotroph) [inline modules/uncertainty/DESCRIPTION:48]: Nitpicking: Can you move these lines up to stay in alphabetical order, please?

File: scripts/Makefile.depends [hunk 6]
```diff
@@ -0,0 +1 @@
+# autogenerated
```
Comment (infotroph) [inline scripts/Makefile.depends:1]: I don't know why this line shows as added here when it's present in develop too, but seems OK

File: CHANGELOG.md [hunk 7]
```diff
@@ -20,6 +20,7 @@ see if you need to change any of these:
   to star PEcAn with https.
 - TRAEFIK_HTTPS_OPTIONS is no longer used, this is the default when you use https.
 - TRAEFIK_HTTP_REDIRECT is no longer used, this is the default when you use https.
+- PEcAn.assim.sequential is renamed to PEcAnAssimSequential
```
[the remark this answers is not in the document]
Comment (helge22a) [reply CHANGELOG.md:23]: Yes I will move this line down

File: modules/uncertainty/DESCRIPTION (hunk 5 above)
Comment (helge22a) [reply modules/uncertainty/DESCRIPTION]: so I am removing the dependency on these packages so they are no longer in the DESCRIPTION file.

File: modules/uncertainty/R/run.sensitivity.analysis.R [hunk 8]
```diff
@@ -133,7 +133,7 @@ run.sensitivity.analysis <- function(settings,plot=TRUE, ensemble.id=NULL, varia
             pdf(fname, height = 12, width = 9)
             ## arrange plots  http://stackoverflow.com/q/10706753/199217
             ncol <- floor(sqrt(length(sensitivity.plots)))
-            print(do.call("grid.arrange", c(sensitivity.plots, ncol=ncol)))
+            print(do.call("gridExtra::grid.arrange", c(sensitivity.plots, ncol=ncol)))
```
Comment (infotroph) [inline modules/uncertainty/R/run.sensitivity.analysis.R:136]:
```suggestion
            print(do.call(gridExtra::grid.arrange, c(sensitivity.plots, ncol=ncol)))
```

(I don't know why do.call accepts non-namespaced function names as character but namespaced ones only as symbols, but at least the fix is easy to implement 🤷‍♂️)

**Review by infotroph — APPROVED** [review]
